utils/linear_probes.py

```diff
@@ -0,0 +1,298 @@
+    
+    # brainstorming cmaps; some to try: with blues and reds: 'PuOr', 'RdBu', 'RdYlBu', 'RdYlGn', 'Spectral', 'coolwarm'   
+    sns.heatmap(scores.T, cmap=cmap, vmin=vmin, vmax=vmax)
+    
+    plt.xticks(np.arange(len(tokens)), tokens, rotation=60) # set x ticks to the actual str tokens
+    plt.gca().set_xticks(np.arange(len(tokens))+0.5, minor=True) # position x ticks in the middle of the cell without extra ticks
+    plt.gca().tick_params(which='major', length=0) # remove major ticks and keep only minor ticks
+    plt.gca().invert_yaxis() # make y axis go from bottom to top
+
+    plt.xlabel('Token')
+    plt.ylabel('Layer')
+    if title is not None:
+        plt.title(title)
+    
+    # plt.yticks(np.arange(len(scores.T))+0.5, range(1, len(scores.T)+1)) # add 1 to every y tick without changing its position
+    plt.yticks(np.arange(len(scores.T)), range(1, len(scores.T)+1)) # add 1 to every y tick without changing its position
+
+    # thin grid lines
+    plt.grid(which='major', color='gray', linestyle='-', linewidth=0.3)
+
+    # # remove every 2nd y tick label
+    for label in plt.gca().yaxis.get_ticklabels()[::2]:
+        label.set_visible(False)
+    for label in plt.gca().yaxis.get_ticklabels():
+        label.set_verticalalignment('bottom')
+        # label.set_position((0, 0.5))
+    for label in plt.gca().xaxis.get_ticklabels():
+        label.set_horizontalalignment('center')
+    
+    # add minor y ticks every 2nd minor tick
+    plt.gca().set_yticks(np.arange(len(scores.T))+0.5, minor=True)
+    # go over ticks and set every 2nd tick to be invisible
+    for tick in plt.gca().yaxis.get_minor_ticks()[::2]:
+        tick.tick1line.set_visible(False)
+        tick.tick2line.set_visible(False)
+        tick.label1.set_visible(False)
+        tick.label2.set_visible(False)
+        
+    plt.tight_layout()
+    
+
+def main():
+    """Example usage of run_q_type and plot_score_grid"""
+    torch.set_grad_enabled(False)
+    torch.set_num_threads(8)
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    seed = 600
+    seed_stage2 = 0
+    np.random.seed(seed)
+    
+    # TODO just load these params from config file
+    data =  get_questions_dataset(
+        seed=seed,
+        seed_stage2=seed_stage2,
+        frac_n_qd1consis= 0.25,
+        frac_n_qd1incons= 0.0,
+        frac_n_qd2consis= 0.0,
+        frac_n_qd2incons= 0.25,
+        frac_n_q= 0.1,
+        frac_n_d1consis= 0.08,
+        frac_n_d2consis= 0.08,
+        frac_n_d3consis= 0.08,
+        frac_n_no_qd_baseline= 0.06,
+        frac_n_q_no_replacement_baseline=0.1,
+        dataset_name='cvdb',
+        train_subset='full',
+        num_ents=4000,
+        entity_association_test_sets=True,
+        multiple_define_tags=False,
+        incontext_defs=False
+    )
+    print(data.keys())
+    
+    q_type = 'born'
+
+    data1 = data['qd1consis']['question']
+    data2 = data['qd2incons']['question']
```

Comment:
do you only compare these?

Reply:
Nope, I have a jupyter notebook with lots of other comparisons